protobuf/lib/src/protobuf/proto3_json.dart

```diff
@@ -415,3 +417,26 @@ void _mergeFromProto3Json(
 
   recursionHelper(json, fieldSet);
 }
+
+void _initializeRequiredFields(_FieldSet fieldSet) {
+  final meta = fieldSet._meta;
+
+  if (!meta.hasRequiredFields) {
+    return;
+  }
+
+  for (final fieldInfo in meta.sortedByTag) {
+    if (!fieldInfo.isRequired) {
+      continue;
+    }
+
+    var value = fieldInfo.makeDefault!();
```

Comment:
I think this probably generates immutable lists and maps, which we should unfreeze below.

Reply:
Ah, but maps and lists cannot be `required`. So I guess this is fine.